pkg/endpoint/bpf.go

```diff
@@ -1146,3 +1145,22 @@ func (e *Endpoint) RequireRouting() (required bool) {
 func (e *Endpoint) RequireEndpointRoute() bool {
 	return e.DatapathConfiguration.InstallEndpointRoute
 }
+
+type linkCheckerFunc func(string) error
+
+// ValidateConnectorPlumbing checks whether the endpoint is correctly plumbed
+// depending on if it is conected via veth or IPVLAN.
+func (e *Endpoint) ValidateConnectorPlumbing(linkChecker linkCheckerFunc) error {
+	if e.HasIpvlanDataPath() {
+		// FIXME: We cannot check whether ipvlan slave netdev exists,
+		// because it requires entering container netns which is not
+		// always accessible (e.g. in k8s case "/proc" has to be bind
+		// mounted). Instead, we check whether the tail call map exists.
+		if _, err := os.Stat(e.BPFIpvlanMapPath()); err != nil {
+			return fmt.Errorf("tail call map for IPvlan unavailable: %s", err)
+		}
+	} else if err := linkChecker(e.ifName); err != nil {
```

Comment:
I suppose it's now possible to pass in a nil linkChecker. I'm not sure that it's worth guarding against that, though... it would be very unexpected (but will happen when we start passing variables about that are the callback).

Reply:
Good point, will add a nil check.

Reply:
Fixed in latest upload.